plugins/differ/nri-differ.go

```diff
@@ -177,74 +177,74 @@ func (p *plugin) differ(apifunc string, pod *api.PodSandbox, container *api.Cont
 	}
 }
 
-func (p *plugin) Synchronize(pods []*api.PodSandbox, containers []*api.Container) ([]*api.ContainerUpdate, error) {
+func (p *plugin) Synchronize(_ context.Context, pods []*api.PodSandbox, containers []*api.Container) ([]*api.ContainerUpdate, error) {
 	if cfg.VerboseLevel > 2 {
 		p.dump("Synchronize", "pods", pods, "containers", containers)
 	}
 
 	return nil, nil
 }
 
-func (p *plugin) Shutdown() {
+func (p *plugin) Shutdown(_ context.Context) {
 	p.dump("Shutdown")
 }
 
-func (p *plugin) RunPodSandbox(pod *api.PodSandbox) error {
+func (p *plugin) RunPodSandbox(_ context.Context, pod *api.PodSandbox) error {
 	p.differ("RunPodSandbox", pod, nil)
 	return nil
 }
 
-func (p *plugin) StopPodSandbox(pod *api.PodSandbox) error {
+func (p *plugin) StopPodSandbox(_ context.Context, pod *api.PodSandbox) error {
 	p.differ("StopPodSandbox", pod, nil)
 	return nil
 }
 
-func (p *plugin) RemovePodSandbox(pod *api.PodSandbox) error {
+func (p *plugin) RemovePodSandbox(_ context.Context, pod *api.PodSandbox) error {
 	p.differ("RemovePodSandbox", pod, nil)
 	return nil
 }
 
-func (p *plugin) CreateContainer(pod *api.PodSandbox, container *api.Container) (*api.ContainerAdjustment, []*api.ContainerUpdate, error) {
+func (p *plugin) CreateContainer(_ context.Context, pod *api.PodSandbox, container *api.Container) (*api.ContainerAdjustment, []*api.ContainerUpdate, error) {
 	p.differ("CreateContainer", pod, container)
 
 	adjust := &api.ContainerAdjustment{}
 
 	return adjust, nil, nil
 }
 
-func (p *plugin) PostCreateContainer(pod *api.PodSandbox, container *api.Container) error {
+func (p *plugin) PostCreateContainer(_ context.Context, pod *api.PodSandbox, container *api.Container) error {
 	p.differ("PostCreateContainer", pod, container)
 	return nil
 }
 
-func (p *plugin) StartContainer(pod *api.PodSandbox, container *api.Container) error {
+func (p *plugin) StartContainer(_ context.Context, pod *api.PodSandbox, container *api.Container) error {
 	p.differ("StartContainer", pod, container)
 	return nil
 }
 
-func (p *plugin) PostStartContainer(pod *api.PodSandbox, container *api.Container) error {
+func (p *plugin) PostStartContainer(_ context.Context, pod *api.PodSandbox, container *api.Container) error {
 	p.differ("PostStartContainer", pod, container)
 	return nil
 }
 
-func (p *plugin) UpdateContainer(pod *api.PodSandbox, container *api.Container) ([]*api.ContainerUpdate, error) {
+func (p *plugin) UpdateContainer(_ context.Context, pod *api.PodSandbox, container *api.Container, r *api.LinuxResources) ([]*api.ContainerUpdate, error) {
```

Comment:
might want to split the LinuxResources change from the ctx change..

Reply:
I split out the update/resource-related changes to two additional separate commits: one to add the missing resource parameter to the stub plugin UpdateContainer handler, and another to update the mocked test plugins and to implement basic tests for UpdateContainers.

@mikebrow I'll file those in a separate PR. I interpreted it as your preference from the above comment, as opposed to having them in separate commits but part of the same PR.

Reply:
splitting the commit is fine.. just need to have a different description in the PR for that additional change..